src/test/collectionHandlingFunctions.php

```diff
@@ -0,0 +1,314 @@
+<?php
+require "../collectionHandlingFunctions.php";
+use PHPUnit\Framework\TestCase;
+
+class collectionHandlingFunctions extends TestCase 
+{
+
+    public function testFailureDescribeCollectionItem_EmptyItem() 
+    {
+        $yarnExample = [];
+        $this->expectException(Exception::class); 
+        describeCollectionItem($yarnExample);
+    }
+    
+    public function testFailureDescribeCollectionItem_undefinedName() {
+        $yarnExample = ["name" => NULL, 
+                        "shade" => "Rainbow", 
+                        "dominantColour" => "purple",
+                        "composition" => "Cashmere",
+                        "yarnType" => "Super Chunky", 
+                        "lengthInMeters" => 100,
+                        "skeinNumber" => 2, 
+                        "project"=> "Cool rainbow hat",
+                        "image" => "blue_scarf.jpg"      
+                    ];
```

Comment:
it is correct to put [ on line 16, then everything starting from the line below, the indentation should be different as well (make sure to change this in all the test functions)

Reply:
done!